### PR TITLE
Fix crash when propagating simulation ownership on deleted objects

### PR DESCRIPTION
**Pull request by AndrewMeadows**
- fix crash when deleting a pile of colliding objects

### CODE REVIEW

**Comment by AndrewMeadows** [comment]
The top of the callstack for this crash mode looks like this:  crash callstack for this mode looks like this:

    interface.exe!PhysicsEngine::bump(ObjectMotionState * motionState) Line 410 C++
    interface.exe!PhysicsEngine::removeObjects(const QVector<ObjectMotionState *> & objects) Line 152 C++
    interface.exe!Application::update(float deltaTime) Line 3110 C++
    interface.exe!Application::idle(unsigned __int64 now) Line 2440 C++
    interface.exe!Application::paintGL() Line 1318 C++

The way to tickle it is to make a pile of dynamic colliding objects and then to delete them all at once.  Sometimes objectB would be deleted before objectA and we would try to "bump" the simulation ownership of invalid objectB.  There is a test script in #7061 that has been able to reproduce this crash at least once, although it appears the success rate is somewhere around one in ten.

Note: this PR depends on PR-7061 since I was hitting that other crash mode before reaching this one.

File: libraries/physics/src/PhysicsEngine.cpp [hunk 1]
```diff
@@ -136,23 +136,19 @@ void PhysicsEngine::addObjectToDynamicsWorld(ObjectMotionState* motionState) {
...
-    btRigidBody* body = object->getRigidBody();
-    assert(body);
-    _dynamicsWorld->removeRigidBody(body);
-}
-
 void PhysicsEngine::removeObjects(const VectorOfMotionStates& objects) {
+    // first bump and prune contacts for all objects in the list
+    for (auto object : objects) {
+        bumpAndPruneContacts(object);
+    }
```
Comment (AndrewMeadows) [inline libraries/physics/src/PhysicsEngine.cpp]: The actual fix for the crash is here: first bump and prune contacts for all objects in the list above, then remove them below.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 00e90f6f7d463f93b196c4f99db1eec954dc83a3 available at: https://gist.github.com/af34a185856059296d01<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7069.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7069.dmg

**Comment by AndrewMeadows** [comment]
The internal asana task for this is here: https://app.asana.com/0/26623005675174/87216130762001

**Comment by bluetoe** [comment]
This still crashes interface for me every once in awhile. using the script from PR-7061 I've crashed on 8th run, then on 10th run, then 12th. Running the script a second time before first set has deleted seems to make crash happen quicker, but not always

**Comment by AndrewMeadows** [comment]
@bluetoe, the question I have is: What does the callstack look like?  There is at least one, maybe two, distinct crash modes that could happen from the same test.  I'm trying to track it down however I'm having trouble reproducing it.  Since I've been testing the Debug build so far I'm now moving to Release to see if that helps.

**Comment by bluetoe** [comment]
I was going to get call stack but found that I needed to update VS first and I got lazy.. but ok. I will try to get it

**Comment by bluetoe** [comment]
pr-7069-crash

**Comment by AndrewMeadows** [comment]
Yup, this still needs work.  I was just able to reproduce the crash on windows and linux using the **Release** build.

**Comment by AndrewMeadows** [comment]
Ok, I give up making the cleanup flow airtight... I added a check for valid **MotionState** when I used to assert.  After that I wasn't able to crash on linux despite much creation/deletion of objects, so this might be fixed.  I will test on windows too.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7aa1d2deeaebec54a229cffbcefc70a070f31b04 available at: https://gist.github.com/af34a185856059296d01<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7069.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7069.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ac60917b546e987e320452a52b742d96db93280e available at: https://gist.github.com/af34a185856059296d01<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7069.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7069.dmg

**Comment by bluetoe** [comment]
That last one seems to have fixed it. I tried 24 runs of the test script and no crash, then tried 12 double click runs, then tried 8 triple click runs and no crash. Although with triple click the physics seem to disappear for some of the blocks. But anyway, 14400 blocks deleted and simulation ownership changed so, much better than before.

**Comment by AndrewMeadows** [comment]
Yup, I did a bunch of testing on Windows and linux simultaneously last night right before I left.  Probably about 15 events, each with 200 entities, and neither crashed.

File: libraries/physics/src/PhysicalEntitySimulation.cpp [hunk 2]
```diff
@@ -167,12 +167,13 @@ void PhysicalEntitySimulation::getObjectsToRemoveFromPhysics(VectorOfMotionState
...
-        _physicalObjects.remove(motionState);
-        result.push_back(motionState);
-        _entitiesToRelease.insert(entity);
+        if (motionState) {
+            _pendingChanges.remove(motionState);
+            _outgoingChanges.remove(motionState);
+            _physicalObjects.remove(motionState);
+            result.push_back(motionState);
+            _entitiesToRelease.insert(entity);
+        }
```
Comment (Atlante45) [inline libraries/physics/src/PhysicalEntitySimulation.cpp]: Should there be an `else if` here?
I noticed when going through the `_entitiesToRelease`, we then add them to `_entitiesToDelete`.
So they would end up being added twice to `_entitiesToDelete`.

Comment (Atlante45) [inline libraries/physics/src/PhysicalEntitySimulation.cpp]: Nvm, I see that's a set.

**Pull request merged**